### PR TITLE
[BUU] Dropdown UI tweaks (tom-select)

### LINKED ISSUE #11518: Dropdown UI tweaks
## Description

**- As a:** enterprise user or super admin
**- On page:** `/admin/products_v3` => this page is behind feature toggle `admin_style_v3`
**- I want to be able to do:** 

- [x] change font size to default (14px)
- [x] when picker is active, but nothing is selected/typed in, change placeholder text to search for <label> (if easy enough to implement)
- [x] when picker is active, but nothing is selected/typed in, change placeholder text color to #BABDBE ($silver-sand)
- [x] when picker is active, change border color to $orient
- [x] when picker is active, add shadow to list container (style added to [Colours guide](https://github.com/openfoodfoundation/openfoodnetwork/wiki/Design-styleguide:-colors-and-shadows))
- [x] add a 4px border to the list (i know this might be painful with the scrollbar. if too complex, let's not worry about it
- [x] for producers and categories, in the list of options, include an All producers, All categories option
- [x] active and hover list items to have same style, see [this prototype](https://www.figma.com/proto/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery---Back-Office?page-id=373%3A2076&type=design&node-id=373-2077&viewport=238%2C463%2C0.23&t=dtlWH3LOdt8Sbqqp-1&scaling=min-zoom&starting-point-node-id=373%3A2077&mode=design) for example
- [x] related to previous, increase thickness of vertical line to 3px
- [x] increase height of each list option to 36px
- [x] add top and bottom margin of 4px to the list 
Screenshot 2023-09-11 at 11 20 43
- [x] after a list item is selected, the field is still active, it should go back to default state (see screenshot below)
Screenshot 2023-09-11 at 11 05 32

#### Additional nice to have requirements
- Handle long content better. (either truncate with ellipsis, or allow field to stretch in height?)
- On all selects: change icon to U+F078 chevron_down
- Search row columns: search input can grow, so we don't have whitespace. dropdowns can sit closer to search button. search input and dropdowns should each have a min-width.

Figma screens are available here >
https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery---Back-Office?type=design&node-id=378-3907&mode=design&t=hqimHkMq8OKx5CII-0

## Acceptance Criteria & Tests

1. Activate feature toggle
2. Log in as enterprise
3. On  `/admin/products_v3` the dropdowns should met the above criterias

Warning: if your user has access to only one producer, you won't see the producer dropdown

### PR DESCRIPTION
**Pull request by dacook**
### What? Why?

- Closes #11518

### What should we test?
With `admin_style_v3` feature toggle,
Check that dropdowns behave as expected, eg in:
* `/admin/products`
* `/admin/orders`
* `/admin/enterprises/*/edit#/address_panel`

Note that some dropdowns still have the older style, they can be changed over in future PRs

#### Single select with search:
Screen Shot 2023-12-15 at 3 50 21 pm
#### Single select with no search:
Screen Shot 2023-12-15 at 3 50 37 pm

### CODE REVIEW

**Comment by dacook** [comment]
@mariocarabotta , when you're in front of a computer, it would be great if you could stage and review this PR

I've done most of what's listed on the issue, but found it fiddly to get it to behave quite right. The main (pre-existing) issue is this scenario:
1. Click on (or tab to) Producers or Categories
2. Select any option
3. Click on the field again (or press down arrow)
* ❌ The previously-selected option is still visible instead of the placeholder, and there's no flashing caret to show that you can type in. 
* ❌ When you do type in, it appears below the previous selection

I think this is probably fixable, I just haven't got there yet. Let me know if you'd like to workshop it, otherwise I can keep working on it.
In the meantime, I can work on the other requirements (which I've added to the bottom of the issue description for transparency).

**Comment by mariocarabotta** [comment]
Screenshot 2023-12-17 at 16 43 01

couple of notes:
- i'd be good if when highlighted, this uses the Orient colour border
- there's a strange double shadow showing when selected, see screenshot above.

Other than that, it looks great! awesome work!

**Review by mkllnk — APPROVED** [review]
Great!

**Comment by dacook** [comment]
> * i'd be good if when highlighted, this uses the Orient colour border
> * there's a strange double shadow showing when selected, see screenshot above.

We found that this only seems to affect Mario's browser (Brave), so won't try and fix it at this stage.

File: app/views/admin/products_v3/_sort.html.haml [hunk 1]
```diff
@@ -6,4 +6,4 @@
         = t(".pagination.clear_search")  
   %form.with-dropdown
     = t(".pagination.per_page.show")
-    = select_tag :per_page, options_for_select([15, 25, 50, 100].collect{|i| [t('.pagination.per_page.per_page', num: i), i]}, pagy.items), data: { reflex: "change->products#change_per_page" }
+    = select_tag :per_page, options_for_select([15, 25, 50, 100].collect{|i| [t('.pagination.per_page.per_page', num: i), i]}, pagy.items), class: "no-input", data: { reflex: "change->products#change_per_page", controller: "tom-select", "tom-select-options-value": '{ "plugins": [] }'}
```
Comment (rioug) [inline app/views/admin/products_v3/_sort.html.haml]: Adding `tom-select` stimulus controller seems unrelated to this commit, why is it needed ?

File: app/webpacker/css/admin_v3/components/tom_select.scss [hunk 2]
```diff
@@ -43,15 +37,23 @@
...
     @include border-radius($border-radius);
-    border: none;
     box-shadow: $shadow-dropdown;
 
-    .ts-dropdown-content {
-      padding: $border-radius 0;
-    }
+    // .ts-dropdown-content {
+    //   padding: $border-radius 0;
+    // }
```
Comment (rioug) [inline app/webpacker/css/admin_v3/components/tom_select.scss:56]: Should be deleted instead of commented out ?

**Review by rioug — COMMENTED** [review]
Nice one :clap:  I just have a couple of comments

File: app/views/admin/products_v3/_sort.html.haml (hunk 1 above)
Comment (dacook) [reply app/views/admin/products_v3/_sort.html.haml]: Sorry, the commit message should have explained better. 
This issue introduces a new design for single-select dropdowns.
We already had some single-select tom-select dropdowns in the admin interface, so the updated design applies to them. 
Tom-select is also added to this select, so we can apply the new design to it as required by the issue. Hope that makes sense.

File: app/webpacker/css/admin_v3/components/tom_select.scss (hunk 2 above)
Comment (dacook) [reply app/webpacker/css/admin_v3/components/tom_select.scss]: Oops, yes I amended the commit but apparently never pushed it up! 
Pushed up now, plus another small commit.

**Comment by mariocarabotta** [comment]
> > * i'd be good if when highlighted, this uses the Orient colour border
> > * there's a strange double shadow showing when selected, see screenshot above.
> 
> We found that this only seems to affect Mario's browser (Brave), so won't try and fix it at this stage.

this was bugging me too much ;D 

I think it can be fixed by changing vertical padding from 0.5rem to 0.6 rem here

.plugin-dropdown_input .dropdown-input {
    background-color: #fff;
    border: 1px solid #f8f9fa;
    box-shadow: none;
    padding: 0.6rem 0.75rem;
}

Screenshot 2024-01-10 at 10 50 47

**Review by rioug — APPROVED** [review]
Nice 1

**Comment by dacook** [comment]
Thanks Mario, that looks good to me. I've updated it, and started deploying to au_staging

**Comment by dacook** [comment]
As this is quite a different component now, and it affects multiple screens, I'm going to request some testing.

**Comment by RachL** [comment]
> Note that some dropdowns still have the older style, they can be changed over in future PRs

@dacook just to be sure, when speaking about "old style" you include placeholder, size and "All" option?

**Comment by dacook** [comment]
> when speaking about "old style" you include placeholder, size and "All" option?

More precisely, some dropdowns use the old "select2" plugin, which I have not restyled yet. These appear with a triangle icon, and a separate search input like this:
Screenshot 2024-01-12 at 8 15 15 am

**Comment by sigmundpetersen** [comment]
Hey @RachL are you choosing the rebase option when updating the branch?
image

**Comment by RachL** [comment]
@sigmundpetersen yes I thought I did, but does not look like it :/

**Comment by filipefurtad0** [comment]
Merging and including it in the release, [as per this discussion](https://openfoodnetwork.slack.com/archives/C02TZ6X00/p1705056263412999?thread_ts=1705055910.596889&cid=C02TZ6X00).
